blobs.go

```diff
@@ -0,0 +1,71 @@
+package distribution
+
+import (
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/docker/distribution/digest"
+	"golang.org/x/net/context"
+)
+
+// Blob describes a readable and seekable binary object. Most registry
+// objects, including layers and manifests can be accessible by digest in a
+// BlobService.
+type Blob interface {
+	Describable
+	io.ReaderAt
+	io.Closer
+
+	Reader() (ReadSeekCloser, error)
```

Comment:
Why not `NewReadSeeker(blob io.ReaderAt) (io.ReadSeekCloser, error)` then every read becomes a ReadAt and the underlying ReadSeeker will need to keep track of the read state.